rdr_service/tools/study_nph_biobank_file_export.py

```diff
@@ -0,0 +1,222 @@
+        with nph_sample_update_dao.session() as session:
+            sample_updates_ = session.query(SampleUpdate).filter(
+                SampleUpdate.rdr_ordered_sample_id.in_(ordered_sample_ids)
+            ).all()
+            sample_updates.extend(sample_updates_)
+    return sample_updates
+
+
+def _compute_crc32c_checksum_using_file_buffer(json_filepath: str) -> int:
+    crc32c_checksum = 0
+    with open(json_filepath, 'r') as json_fp:
+        file_buffer = json_fp.read(FILE_BUFFER_SIZE_IN_BYTES)
+        while file_buffer:
+            crc32c_checksum = crc32(file_buffer.encode(), crc32c_checksum)
+            file_buffer = json_fp.read(FILE_BUFFER_SIZE_IN_BYTES)
+        return crc32c_checksum
+
+
+def _create_biobank_file_export_reference(json_filepath: str) -> BiobankFileExport:
+    crc32c_checksum = _compute_crc32c_checksum_using_file_buffer(json_filepath)
+    biobank_file_export_dao = NphBiobankFileExportDao()
+    biobank_file_export_params = {
+        "file_name": json_filepath,
+        "crc32c_checksum": str(crc32c_checksum)
+    }
+    biobank_file_export = BiobankFileExport(**biobank_file_export_params)
+    return biobank_file_export_dao.insert(biobank_file_export)
+
+
+def _create_sample_export_references_for_sample_updates(
+    biobank_file_export_id: int, sample_updates: Iterable[SampleUpdate]
+):
+    sample_export_dao = NphSampleExportDao()
+    for sample_update in sample_updates:
+        sample_export_params = {
+            "export_id": biobank_file_export_id,
+            "sample_update_id": sample_update.id
+        }
+        sample_export = SampleExport(**sample_export_params)
+        sample_export_dao.insert(sample_export)
+
+
+def main():
+    orders_file_drop: List[Dict[str, Any]] = []
+    orders: Iterable[Order] = get_orders_created_or_modified_in_last_n_hours(hours=24)
+    grouped_orders: Dict[int, List[Dict[str, Any]]] = defaultdict(list)
+    for order in orders:
+        finalized_site = order.finalized_site
+        nph_module_id = _get_parent_study_category(_get_parent_study_category(order.category_id).id)
+        participant_id = order.participant_id
+        grouped_orders[(finalized_site, nph_module_id.name, participant_id)].append(order)
+
+    for (finalized_site, nph_module_id, participant_id), orders in grouped_orders.items():
+        rdr_participant_summary: RdrParticipantSummary = (
+            _get_rdr_participant_summary_for_nph_partipant(order.participant_id)
+        )
+        json_object = {
+            "clientID": finalized_site,
+            "studyID":  nph_module_id,
+            "participantID": participant_id,
+            "gender": rdr_participant_summary.sexId,
+            "ai_an_flag": "Y" if rdr_participant_summary.aian else "N",
+            "collections": _convert_orders_to_collections(orders, rdr_participant_summary)
+        }
+        orders_file_drop.append(json_object)
+
+    today_dt_ts = datetime.now().strftime("%Y_%m_%d_%H_%M_%S")
```

Comment:
We should use UTC time in the file name. I think the server's local time is set to UTC anyway, and maybe it wouldn't matter too much. But it could get confusing if the timezone depends on who ran the script.